src/crimson/os/seastore/seastore_types.h

```diff
@@ -62,44 +82,33 @@ struct segment_id_t {
     0xF << (std::numeric_limits<internal_segment_id_t>::digits - DEVICE_ID_LEN_BITS);
   // default internal segment id
   static constexpr internal_segment_id_t DEFAULT_INTERNAL_SEG_ID =
-    std::numeric_limits<internal_segment_id_t>::max() - 1;
+    (std::numeric_limits<internal_segment_id_t>::max() >> 1) - 1;
 
   internal_segment_id_t segment = DEFAULT_INTERNAL_SEG_ID;
 
   constexpr segment_id_t(uint32_t encoded) : segment(encoded) {}
 public:
   constexpr static segment_id_t make_max() {
-    return std::numeric_limits<internal_segment_id_t>::max();
+    return std::numeric_limits<internal_segment_id_t>::max() >> 1;
   }
```

Comment:
Where do we use segment_id_t::make_null(), make_zero(), make_fake() outside of the context of a paddr_t?  Can we move these into paddr_t as well?

Reply:
I think we need to keep these values to check if segment id is NULL or ZERO or FAKE, not paddr_t. See below.

1. NULL_SEG_ID in get_next_gc_target()
`if (id != NULL_SEG_ID) {....}`

2. ZERO_SEG_ID in get_segment()
`  return get_segment_ret(
    get_segment_ertr::ready_future_marker{},
    ZERO_SEG_ID);
`
3. FAKE_SEG_ID in test_btree_lba_manager.cc---moreover, offset is needed at here.
`return make_fake_paddr(next_off);`

Thought?

Reply:
1. Yeah, NULL_SEG_ID is ok.
2. should actually be NULL_SEG_ID (it's also impossible as it's after an assert).
3. means we need a fake paddr, not a fake segment?

Reply:
Hmm, I guess fake segment_id_t is ok -- it's just a test thing.  We can just remove zero then.